File: backend/App/Core/HttpStatus.php
<?php

declare(strict_types=1);

namespace App\Core;

enum HttpStatus: int
{
    case Ok = 200;
    case Created = 201;
    case NoContent = 204;
    case BadRequest = 400;
    case Unauthorized = 401;
    case Forbidden = 403;
    case NotFound = 404;
    case UnprocessableEntity = 422;
    case InternalServerError = 500;

    public function response(): void
    {
        http_response_code($this->value);
    }
}

File: tests/http_status.py
from enum import IntEnum


class HttpStatus(IntEnum):
    """
    Enum for common status used in http responses.
    """

    OK = 200
    NO_CONTENT = 204
    NOT_FOUND = 404
    INTERNAL_SERVER_ERROR = 500

    def __init__(self, status: int) -> None:
        self.response = f"{status}; charset=UTF-8"
